NeuroID/DataStore.swift

```diff
@@ -28,6 +29,24 @@ public enum DataStore {
     }
 
     static func cleanAndStoreEvent(screen: String, event: NIDEvent, storeType: String) {
+        // If we hit a low memory event, drop events and early return.
+        if (NeuroID.lowMemory) {
+            return
+        }
+        // If queue has more than 2000 events, send a queue full event and return
```

Comment:
How are we counting events/queueEvents in android?  both can collect 2k or 2k summed across both?

Also, is it necessary to track storeType with a full event?  Any reason not to keep it simple and just add it to and check for it in events?

Reply:
On the Android side, the 2000 count limit applies to both the queuedEvents and events. When saveAndClearAllQueuedEvents() is called, this dumps all queuedEvents into the events list.  QueuedEvents is not counted until saveAndClearAllQueuedEvents() is called. There is no limit on how many queuedEvents can be stored.

Reply:
Android logic here...looks like queued buffer is never checked. @kishisaka-nid can you confirm?

https://github.com/Neuro-ID/neuroid-android-sdk/blob/5b89b63d78d7a586667a2a7177e5255900813d1d/NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt#L136

https://github.com/Neuro-ID/neuroid-android-sdk/blob/5b89b63d78d7a586667a2a7177e5255900813d1d/NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt#L744

Reply:
Yes this is correct. The queuedEvents lists has no limit check. Once it is saved (flushed into the events list) it is counted as part of the 2000 events limit. The flush occurs on start() or startSession().

Reply:
It is easier to add together the two lists and check if this exceeds 2k events before adding an event to either list.

Reply:
I mean, I can change the implementation but these will both effectively do the same thing...

Reply:
If changing the implementation is not going to change the behavior then suggest you not change it.

Reply:
Let's cap the total to 2k and just track the event in the events list regardless of source event.

@kishisaka-nid can you add the combined cap to android?

Reply:
@quacke-nid will do.